src/addon-manager-proxy.ts

```diff
@@ -719,9 +816,12 @@ class AddonManagerProxy extends EventEmitter {
    *
    * Unloads the plugin, and tells the server about it.
    */
-  unloadPlugin() {
-    if (this.pluginClient.ipcProtocol === 'inproc') {
-      if (this.onUnload !== null) {
+  unloadPlugin(): void {
+    const pluginClientWithIpc =
+      <{ipcProtocol: string}><unknown> this.pluginClient;
+
+    if (pluginClientWithIpc.ipcProtocol === 'inproc') {
```

Comment:
We can actually drop the `inproc` nonsense, as it's no longer used by the gateway.

Reply:
Done